webhook/webhook.go

```diff
@@ -0,0 +1,104 @@
+package webhook
+
+import (
+  "crypto/hmac"
+  "crypto/sha256"
+  "encoding/hex"
+  "net/http"
+  "fmt"
+  "errors"
+  "encoding/json"
+  "bytes"
+  "encoding/gob"
+
+  log "github.com/gophish/gophish/logger"
+)
+
+const (
+  DefaultTimeoutSeconds = 10
+  MinHttpStatusErrorCode = 400
+  SignatureHeader = "X-Gophish-Signature"
+)
+
+
+//TODO
+
+type Sender interface {
+  Send(url string, secret string, data interface{}) error
+}
+
+type defaultSender struct {
+  client *http.Client
+} 
+
+var senderInstance Sender
+func newDefaultSender() Sender {
+  sn := &defaultSender{}
+  sn.client = &http.Client{
+    Timeout: DefaultTimeoutSeconds,
```

Comment:
Another thing we might consider is adding a configuration which refuses to follow redirects, though I could see this being a stretch goal for later.

Reply:
Noted